packages/ui/src/search/simpleSearch.js

```diff
@@ -9,69 +10,88 @@ const SearchWrapper = styled.div`
   align-items: center;
 `
 
-
 const SearchInput = styled.input`
   height: 25px;
   width: 210px;
   border: 0;
   border-bottom: 1px solid #000;
   background-color: transparent;
-  /*margin-top: 2px;*/
-  /*margin-right: 100px;*/
   text-indent: 5px;
   -webkit-transition: width 0.4s ease-in-out;
   transition: width 0.4s ease-in-out;
 `
 
-const ClearSearch = styled.button`
+const ClearSearchButton = styled.button`
   margin-left: 5px;
   height: 20px;
-
 `
 
-export const Search = ({
-  listName,
-  options,
-  placeholder,
-  onChange,
-  reference,
-}) => {
-  let elem
-  return (
-    <SearchWrapper>
+const SEARCH_KEYBOARD_SHORTCUTS = ['command+f', 'meta+s']
+
+export class Search extends Component {
+  static propTypes = {
+    placeholder: PropTypes.string,
+    onChange: PropTypes.func.isRequired,
+    withKeyboardShortcuts: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    placeholder: 'Search',
+    withKeyboardShortcuts: false,
+  }
+
+  state = {
+    value: '',
+  }
+
+  componentDidMount() {
+    if (this.props.withKeyboardShortcuts) {
+      Mousetrap.bind(SEARCH_KEYBOARD_SHORTCUTS, (e) => {
+        if (this.input) {
+          e.preventDefault()
+          this.input.focus()
+        }
+      })
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.boundKeyboardShortcuts) {
+      Mousetrap.unbind(SEARCH_KEYBOARD_SHORTCUTS)
+    }
+  }
+
+  onChange = (e) => {
+    const value = e.target.value
+    this.setState({ value })
+    this.props.onChange(value)
+  }
+
+  onClear = () => {
+    this.setState({ value: '' })
+    this.props.onChange('')
+  }
+
+  inputRef = (el) => {
+    this.input = el
+  }
+
+  render() {
+    return (
+      <SearchWrapper>
         <SearchInput
-          type="text"
-          name="search"
           autoComplete="off"
-          placeholder={placeholder}
-          ref={e1 => {
-            elem = e1
-          }}
-          list={listName}
-          onChange={(event) => {
-            event.preventDefault()
-            onChange(event.target.value)
-          }}
+          onChange={this.onChange}
+          placeholder={this.props.placeholder}
+          innerRef={this.inputRef}
+          type="text"
+          value={this.state.value}
         />
-        {/* <datalist id={listName}>
-          {options.map(item => <option value={item} />)}
-        </datalist> */}
```

Comment:
Since the `datalist` element wasn't rendered, the `options` and `listName` props for this component were unused.